### DISCUSSION TITLE
Zero-pause GC implementation

### DISCUSSION THREAD

**Discussion by fabianoliver** [dotnet/runtime#74331] [Q&A]
I hope this is the right place to ask this. This question has been on my mind out of sheer curiosity for a while.

Predictability of latency is quite a concern for various applications. In such cases, people tend to either go out of their way with oftentimes rather.. hacky solutions to partially mitigate this issue, or at times avoid C# (and many other GC'ed languages) entirely.
At the same time, the concept of pause-free collectors seems to be theoretically possible. Java has a fairly popular implementation with Azul's C4, and as far as I remember the outline of their approach has been published as well.

I'm curious as to why this doesn't seem to exist in the C# ecosystem. I would _imagine_ there being significant economic interest behind this, given several industries from finance all the way to the gaming industry would have significant stakes.

Is there anything specific to dotnet/the CLR that would render an implementation impossible?
Or has there simply not been enough interest/incentivization for this to be implemented, given it's of course a very hard thing to get right? Has it ever been discussed by the team - and if so, were there any conclusions that you'd be willing to share - both in terms of whether it's feasible at all, as well as the likelihood of a standard implementation?

**Comment by svick** [comment]
Previous discussion: https://github.com/dotnet/runtime/issues/49059.

**Reply by danmoseley** [reply]
@Dotnet/area-gc-coreclr

**Reply by danmoseley** [reply]
Hmm, that doesn't exist - probably should (@mangod9?)
@Maoni0  then

**Reply by teo-tsirpanis** [reply]
BTW it's `@dotnet/gc`.
